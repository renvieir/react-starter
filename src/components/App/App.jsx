import React, { Component } from 'react'
import style from './App.css' // eslint-disable-line

export default class App extends Component {
  render () {
    return (
      <div styleName='style.App'>
        <header styleName='style.App-header'>
          <h1 styleName='style.App-title'>Welcome to React</h1>
        </header>
        <p styleName='style.App-intro'>
          To get started, edit <code>src/components/App.jsx</code> and save to reload.
        </p>
      </div>
    )
  }
}
