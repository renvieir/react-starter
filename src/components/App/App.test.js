import React from 'react'
import App from './App.jsx'
import { shallow } from 'enzyme'

describe('App', () => {
  it('it should render correctly', () => {
    const component = shallow(<App />)
    expect(component).toMatchSnapshot()
  })
})
