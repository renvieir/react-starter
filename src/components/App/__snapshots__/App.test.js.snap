// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App it should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="src-components-App-___App__App-header"
        >
          <h1
            className="src-components-App-___App__App-title"
          >
            Welcome to React
          </h1>
        </header>,
        <p
          className="src-components-App-___App__App-intro"
        >
          To get started, edit 
          <code>
            src/App.js
          </code>
           and save to reload.
        </p>,
      ],
      "className": "src-components-App-___App__App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="src-components-App-___App__App-title"
          >
            Welcome to React
          </h1>,
          "className": "src-components-App-___App__App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to React",
            "className": "src-components-App-___App__App-title",
          },
          "ref": null,
          "rendered": "Welcome to React",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "To get started, edit ",
            <code>
              src/App.js
            </code>,
            " and save to reload.",
          ],
          "className": "src-components-App-___App__App-intro",
        },
        "ref": null,
        "rendered": Array [
          "To get started, edit ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "src/App.js",
            },
            "ref": null,
            "rendered": "src/App.js",
            "type": "code",
          },
          " and save to reload.",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="src-components-App-___App__App-header"
          >
            <h1
              className="src-components-App-___App__App-title"
            >
              Welcome to React
            </h1>
          </header>,
          <p
            className="src-components-App-___App__App-intro"
          >
            To get started, edit 
            <code>
              src/App.js
            </code>
             and save to reload.
          </p>,
        ],
        "className": "src-components-App-___App__App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="src-components-App-___App__App-title"
            >
              Welcome to React
            </h1>,
            "className": "src-components-App-___App__App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to React",
              "className": "src-components-App-___App__App-title",
            },
            "ref": null,
            "rendered": "Welcome to React",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "To get started, edit ",
              <code>
                src/App.js
              </code>,
              " and save to reload.",
            ],
            "className": "src-components-App-___App__App-intro",
          },
          "ref": null,
          "rendered": Array [
            "To get started, edit ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "src/App.js",
              },
              "ref": null,
              "rendered": "src/App.js",
              "type": "code",
            },
            " and save to reload.",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
